.github/workflows/build_deploy_exm_fe.yml

```diff
@@ -0,0 +1,100 @@
+name: Build and Deploy EXM FE
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build:
+    name: Build Angular Application
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+
+    - name: Download and install Maven
+      run: sudo apt-get update && sudo apt-get install -y maven
+
+    - name: Build Angular application using Maven
+      run: mvn clean install
+      working-directory: expenses-manager-frontend
+
+    - name: Upload build artifacts
+      uses: actions/upload-artifact@v2
+      with:
+        name: frontend-build
+        path: expenses-manager-frontend/target/frontend
+
+  deploy:
+    name: Deploy to VM
+    runs-on: ubuntu-latest
+    needs: build
+    environment: deploy
+    if: github.event_name == 'push' || github.event_name == 'workflow_dispatch' || github.event_name == 'pull_request'
```

Comment:
Probabil nu vrem sa facem deploy la pull request, deci merge scoasa asta cu pull_request de aici, dar sa facem deploy cu workflow dispatch sau push in main branch